api-review-trigger workflow fails with exit(2) if comment contains single quotes (`'`)

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->

api-review-trigger workflow fails with exit(2) if comment contains single quotes (`'`)

**To Reproduce**

See workflow execution in https://github.com/hdamker/ReleaseManagement/actions/runs/16250408886:

/home/runner/work/_temp/3ecc9fd1-7c36-4315-916a-2579e237c3cc.sh: line 38: unexpected EOF while looking for matching `)'
💬 Comment context: Issue #32
Error: Process completed with exit code 2.

The issue occurs because:

- The comment contains single quotes (apostrophes) in words like "doesn't" and "I'm"
- GitHub Actions substitutes ${{ toJSON(github.event.comment.body) }} directly into the bash script
- The script uses single quotes around this substitution: echo '${{ ... }}'
- When the comment body contains single quotes, it breaks the bash single-quoted string syntax

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

The "Check Comment Trigger" should exit gracefully independent of the content of the issue/PR comment.

**Additional context**
<!-- Add any other context about the problem here. -->

Proposed fix:

1. Added an env section to the step that sets COMMENT_BODY as an environment variable
2. Changed the bash script to use this environment variable instead of the inline substitution

--- a/api-review/workflows/api-review-trigger.yml
+++ b/api-review/workflows/api-review-trigger.yml
@@ -63,6 +63,8 @@ jobs:
     steps:
       - name: Check Comment Trigger
         id: check
+        env:
+          COMMENT_BODY: ${{ toJSON(github.event.comment.body) }}
         run: |
           # Check if this is a comment event
           if [[ "${{ github.event_name }}" != "issue_comment" ]]; then
@@ -81,7 +83,7 @@ jobs:
           fi
           
           # Use jq to safely extract comment body and get first line only
-          COMMENT_FIRST_LINE=$(echo '${{ toJSON(github.event.comment.body) }}' | jq -r '. | split("\n")[0]')
+          COMMENT_FIRST_LINE=$(echo "$COMMENT_BODY" | jq -r '. | split("\n")[0]')
           
           echo "💬 Comment first line: '$COMMENT_FIRST_LINE'"
           

